bin/yarn2nix.js

```diff
@@ -23,46 +24,70 @@ Options:
 `
 
 const HEAD = `
-{fetchurl, linkFarm}: rec {
+{fetchurl, linkFarm, fetchgit, runCommandNoCC, gnutar}: rec {
   offline_cache = linkFarm "offline" packages;
   packages = [
 `.trim();
 
 ////////////////////////////////////////////////////////////////////////////////
 
-function generateNix(lockedDependencies) {
-  let found = {};
-
-  console.log(HEAD)
-
-  for (var depRange in lockedDependencies) {
-    let dep = lockedDependencies[depRange];
+async function fetchgit(url, sha1) {
+  let file_name = path.basename(url);
+  let url_for_git = url.replace(/^git\+/, "")
+  const {error, stdout, stderr} = await execFile("nix-prefetch-git", [url_for_git, sha1], {});
```

Comment:
this means that using IDF + nix sandboxing won't work if git dependencies are introduced right?

Reply:
`builtins.fetchGit` is must-need - it allows to download private repositories